pkg/identity-manager/casbin_entity_adapter.go

```diff
@@ -0,0 +1,72 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package identitymanager
+
+import (
+	"fmt"
+
+	"github.com/casbin/casbin/model"
+	"github.com/casbin/casbin/persist"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/entity-store"
+)
+
+type CasbinEntityAdapter struct {
+	store entitystore.EntityStore
+}
+
+func NewCasbinEntityAdapter(store entitystore.EntityStore) *CasbinEntityAdapter {
+	return &CasbinEntityAdapter{store: store}
+}
+
+func (a *CasbinEntityAdapter) LoadPolicy(model model.Model) error {
+	var policies []*Policy
+
+	// Policy status like CREATING, DELETING doesn't have any effect on
+	opts := entitystore.Options{
+		Filter: entitystore.FilterExists(),
+	}
+
+	log.Debug("Reloading policies")
+	err := a.store.List(IdentityManagerFlags.OrgID, opts, &policies)
+	for _, policy := range policies {
+		// Casbin authorization rules are of the form (subject, resource, action) and hence the need to iterate over all rule fields.
+		for _, rule := range policy.Rules {
```

Comment:
That's a lot of `for`s! :)

Reply:
I KNEW someone would say that, anyway I will fix it with some memoization 📝